InterfacePython/TasmanianParticleSwarm.py

```diff
@@ -0,0 +1,264 @@
+# FROM OR ARISING OUT OF, IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+
+from ctypes import c_char_p, c_int, c_bool, c_double, c_void_p, POINTER, cdll, cast, CFUNCTYPE
+from numpy.ctypeslib import as_ctypes, ndpointer
+import numpy as np
+import sys
+
+from TasmanianConfig import __path_libdream__
+from TasmanianConfig import TasmanianInputError as InputError
+
+import TasmanianDREAM as DREAM
+
+c_double_p        = POINTER(c_double)
+np_double_arr     = ndpointer(c_double, flags="C_CONTIGUOUS")
+type_optim_obj_fn = CFUNCTYPE(None, c_int, c_int, c_double_p, c_double_p)
+type_optim_dom_fn = CFUNCTYPE(c_bool, c_int, c_double_p)
+type_dream_random = CFUNCTYPE(c_double)
+
+pLibDTSG = cdll.LoadLibrary(__path_libdream__)
+
+pLibDTSG.tsgParticleSwarmState_Construct.restype = c_void_p
+pLibDTSG.tsgParticleSwarmState_Construct.argtypes = [c_int, c_int]
+
+pLibDTSG.tsgParticleSwarmState_Destruct.argtypes = [c_void_p]
+
+pLibDTSG.tsgParticleSwarmState_GetNumDimensions.restype = c_int
+pLibDTSG.tsgParticleSwarmState_GetNumDimensions.argtype = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_GetNumParticles.restype = c_int
+pLibDTSG.tsgParticleSwarmState_GetNumParticles.argtype = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_GetParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetParticleVelocities.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetBestParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetBestPosition.argtypes = [c_void_p, np_double_arr]
+
+pLibDTSG.tsgParticleSwarmState_SetParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_SetParticleVelocities.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_SetBestParticlePositions.argtypes = [c_void_p, np_double_arr]
+
+pLibDTSG.tsgParticleSwarmState_ClearBestParticles.argtypes = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_ClearCache.argtypes = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_InitializeParticlesInsideBox.argtypes = [c_void_p, np_double_arr, np_double_arr, c_char_p, c_int, type_dream_random]
+pLibDTSG.tsgParticleSwarm.argtypes = [type_optim_obj_fn, c_int, type_optim_dom_fn, c_void_p, c_double, c_double, c_double, c_char_p, c_int, type_dream_random]
+
+class ParticleSwarmState:
+    '''
+    Wrapper to class TasOptimization::ParticleSwarmState
+    '''
+    def __init__(self, iNumDimensions, iNumParticles):
+        '''
+        Constructs a new state with a given number of dimensions and particles.
+
+        iNumDimensions : positive integer indicating the number of dimensions.
+        iNumParticles  : positive integer indicating the number of particles.
+        '''
+        self.TasmanianParticleSwarmState = True
+        self.pStatePntr = pLibDTSG.tsgParticleSwarmState_Construct(iNumDimensions, iNumParticles)
```

Comment:
Whatever the `c_void_p` issue is here, the proper solution is to change this to:
```
self.pStatePntr = c_void_p( pLibDTSG.tsgParticleSwarmState_Construct(iNumDimensions, iNumParticles) )
```
then you don't have to type `c_void_p` in every call down the line.